packages/worker/src/interceptors.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Definitions for Activity interceptors.
+ *
+ * (The Worker also accepts Workflow interceptors but those are passed as module names)
+ *
+ * @module
+ */
+
+import { Next, Headers } from '@temporalio/workflow';
+import { Context as ActivityContext } from '@temporalio/activity';
+
+/** Input for ActivityInboundCallsInterceptor.execute */
+export interface ActivityExecuteInput {
+  readonly args: unknown[];
+  readonly headers: Headers;
+}
+
+/**
+ * Implement any of these methods to intercept Activity inbound calls
+ */
+export interface ActivityInboundCallsInterceptor {
+  /**
+   * Called when Activity function is executed
+   *
+   * @return result of Activity function
+   */
+  execute?: (input: ActivityExecuteInput, next: Next<ActivityInboundCallsInterceptor, 'execute'>) => Promise<unknown>;
+}
+
+/**
+ * A function that takes Activity Context and returns an interceptor
+ */
+export interface ActivityInboundCallsInterceptorFactory {
+  (ctx: ActivityContext): ActivityInboundCallsInterceptor;
+}
+
+/**
+ * Structure for passing in Worker interceptors via {@link WorkerOptions}
+ */
+export interface WorkerInterceptors {
+  /**
+   * List of factory functions returning {@link ActivityInboundCallsInterceptor}s
+   */
+  activityInbound?: ActivityInboundCallsInterceptorFactory[];
+  /**
+   * List of modules to search for interceptors in
+   */
+  workflowModules?: string[];
```

Comment:
This is b/c interceptors are defined alongside workflows correct? Maybe worth mentioning that here